Formula/httpd.rb

```diff
@@ -0,0 +1,132 @@
+
+  depends_on "openssl"
+  depends_on "pcre"
+  depends_on "apr-util"
+  depends_on "nghttp2" => :optional
+
+  if build.with?("mpm-worker") && build.with?("mpm-event")
+    raise "Cannot build with both worker and event MPMs, choose one"
+  end
+
+  def install
+    # point config files to opt_prefix instead of the version-specific prefix
+    inreplace "Makefile.in",
+      '#@@ServerRoot@@#$(prefix)#', '#@@ServerRoot@@'"##{opt_prefix}#"
+
+    # fix non-executable files in sbin dir (for brew audit)
+    inreplace "support/Makefile.in",
+      "$(DESTDIR)$(sbindir)/envvars", "$(DESTDIR)$(sysconfdir)/envvars"
+    inreplace "support/Makefile.in",
+      "envvars-std $(DESTDIR)$(sbindir);", "envvars-std $(DESTDIR)$(sysconfdir);"
+    inreplace "support/apachectl.in",
+      "@exp_sbindir@/envvars", "#{etc}/apache2/2.4/envvars"
+
+    # install custom layout
+    (buildpath/"config.layout").unlink
+    (buildpath/"config.layout").write <<-EOS.undent
+      <Layout Homebrew>
+          prefix:        #{prefix}
+          exec_prefix:   ${prefix}
+          bindir:        ${exec_prefix}/bin
+          sbindir:       ${exec_prefix}/bin
+          libdir:        ${exec_prefix}/lib
+          libexecdir:    ${exec_prefix}/libexec
+          mandir:        #{man}
+          sysconfdir:    #{etc}/apache2/2.4
+          datadir:       #{var}/www
+          installbuilddir: ${prefix}/build
+          errordir:      ${datadir}/error
+          iconsdir:      ${datadir}/icons
+          htdocsdir:     ${datadir}/htdocs
+          manualdir:     ${datadir}/manual
+          cgidir:        #{var}/apache2/cgi-bin
+          includedir:    ${prefix}/include/httpd
+          localstatedir: #{var}/apache2
+          runtimedir:    #{var}/run/apache2
+          logfiledir:    #{var}/log/apache2
+          proxycachedir: ${localstatedir}/proxy
+      </Layout>
+    EOS
+
+    args = %W[
+      --enable-layout=Homebrew
+      --enable-mods-shared=all
+      --enable-unique-id
+      --enable-ssl
+      --enable-dav
+      --enable-cache
+      --enable-logio
+      --enable-deflate
+      --enable-cgi
+      --enable-cgid
+      --enable-suexec
+      --enable-rewrite
+      --with-apr=#{Formula["apr"].opt_prefix}
+      --with-apr-util=#{Formula["apr-util"].opt_prefix}
+      --with-pcre=#{Formula["pcre"].opt_prefix}
+      --with-ssl=#{Formula["openssl"].opt_prefix}
+      --with-z=#{Formula["zlib"].opt_prefix}
+      --with-port=8080
+      --with-sslport=8443
+      --with-mpm=event
+    ]
+
+    if build.with? "nghttp2"
+      args << "--enable-http2"
+      args << "--with-nghttp2=#{Formula["nghttp2"].opt_prefix}"
+    end
+
+    (etc/"apache2/2.4").mkpath
+
+    system "./configure", *args
+    system "make"
+    system "make", "install"
+
+    (var/"apache2/log").mkpath
+    (var/"apache2/run").mkpath
+
+    touch("#{var}/log/apache2/access_log")
+    touch("#{var}/log/apache2/error_log")
+  end
+
+  plist_options :manual => "apachectl start"
```

Comment:
should we use the full opt path?

Reply:
Should be in the `PATH` by default so I'm fine with it as-is if you are.

Reply:
ok